### PR TITLE
Fixed CurlMultiHandler Blocking Issue

### PR DESCRIPTION
**Pull request by shtse8**
I am trying to fix this issue https://github.com/guzzle/guzzle/issues/1127

1. Tick() should not block and should not sleep, it should return immediately without waiting.
2. select_timeout changed to 0. return immediately if no message. wait for next tick.
3. if select returns 0 or -1. do nothing and wait for next tick. no need to wait inside for -1. we should wait for next tick.

### CODE REVIEW

**Comment by shtse8** [comment]
For those who are using Curl 7.47.0 with proxy and multi.
Please upgrade to Curl 7.55.1 with following commands:
```shell
sudo add-apt-repository ppa:jonathonf/curl
sudo apt-get update
```


https://curl.haxx.se/changes.html
image

File: src/Handler/CurlMultiHandler.php [hunk 1]
```diff
@@ -95,17 +95,11 @@ public function tick()
...
         P\queue()->run();
 
-        if ($this->active &&
-            curl_multi_select($this->_mh, $this->selectTimeout) === -1
-        ) {
-            // Perform a usleep if a select returns -1.
-            // See: https://bugs.php.net/bug.php?id=61141
-            usleep(250);
+        $cms = curl_multi_select($this->_mh, $this->selectTimeout);
+        if ($cms != 0) {
```
Comment (gregurco) [inline src/Handler/CurlMultiHandler.php]: Wrong comparison. Read docs: http://php.net/manual/en/function.curl-multi-select.php
```
On failure, this function will return -1
```
So, in this case you will do same action when it's failure and success. Also variable name is not clear. I think `$numberOfDescriptors` will be more cleaer. Finally it's better to write `$numberOfDescriptors >= 0`

**Comment by mtdowling** [comment]
This will still block, and I think this change will cause 100% CPU utilization.

**Comment by shtse8** [comment]
@mtdowling I have tested and my project is running without any blocking and CPU issue for 3 months already.

**Pull request closed without merging**